engine/src/main/java/org/terasology/internet/internal/InternetManagerImpl.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.internet.internal;
+
+import com.google.common.collect.Maps;
+import org.terasology.internet.InternetManager;
+import org.terasology.internet.TCPSocket;
+import org.terasology.naming.Name;
+import org.terasology.registry.DynamicInstanceProvider;
+
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Map;
+import java.util.TreeSet;
+
+/**
+ * @author soniex2
+ */
+public class InternetManagerImpl implements InternetManager, DynamicInstanceProvider<InternetManager> {
```

Comment:
There are 2 reasons why I did it this way.

1. Good luck doing it any other way.
2. Uh I forgot the other reason but this feels more natural. We still want a global instance that everyone can use, even if that global instance does nothing, because other parts of the engine can see and access it, and we need something to return from CoreRegistry.get().